org.lflang/src/org/lflang/generator/ts/TSGenerator.kt

```diff
@@ -383,15 +382,92 @@ class TSGenerator(
         serializer: SupportedSerializers
     ): String {
         return with(PrependOperator) {"""
-        // FIXME: For now assume the data is a Buffer, but this is not checked.
-        // Replace with ProtoBufs or MessagePack.
+        |// FIXME: For now assume the data is a Buffer, but this is not checked.
+        |// Replace with ProtoBufs or MessagePack.
+        |// generateNetworkReceiverBody
         |if (${action.name} !== undefined) {
-        |    ${receivingPort.container.name}.${receivingPort.variable.name} = 
-        |    ${action.name}; // defaults to utf8 encoding
+        |    ${receivingPort.container.name}.${receivingPort.variable.name} = ${action.name}.toString(); // defaults to utf8 encoding
         |}
         """.trimMargin()}
     }
 
+    /**
```

Comment:
Perhaps these functions are better put in a `TSFedGenerator.kt`.

Reply:
These are functions that need to be overridden, so I put them in TSGenerator which inherits GeneratorBase. Do you have any specific suggestion?

Reply:
I see. You're right. It is unfortunate that these functions should end up in here. Perhaps we should rethink this design altogether and move these functions out of `GeneratorBase`

Reply:
Yes, I think this is a symptom of our monolithic generator base. I had similar issues in the C++ target, for instance with the methods used for generating after delays.